### PR TITLE
Update units.json | added units for gravity, spm and frequency

### PR DESCRIPTION
**Pull request by hanish2712**

### CODE REVIEW

File: versions/v1/units.json [hunk 1]
```diff
@@ -10494,5 +10494,124 @@
...
+      "Hecto-Hertz",
+      "hHz"
+    ],
+    "symbol": "hHz",
+    "conversion": {
+      "multiplier": 100.0,
+      "offset": 0.0
+    },
+    "source": "qudt.org",
+    "sourceReference": "https://qudt.org/vocab/unit/HZ"
```
Comment (evertoncolling) [inline versions/v1/units.json:10565]: You cannot link hecto hertz to the qudt reference for hertz. In this case, we can accept the following syntax:

```suggestion
    "source": "Custom based on qudt.org - https://qudt.org/vocab/unit/HZ",
    "sourceReference": null
```

File: versions/v1/units.json [hunk 2]
```diff
@@ -10494,5 +10494,124 @@
...
+      "Degree API",
+      "deg API"
+    ],
+    "symbol": "API",
+    "conversion": {
+      "multiplier": 1.0,
+      "offset": 0.0
+    },
+    "source": "qudt.org",
+    "sourceReference": "https://qudt.org/vocab/quantitykind/Gravity_API"
```
Comment (evertoncolling) [inline versions/v1/units.json:10592]: `sourceReference` expects a reference to a unit not a quantity.

```suggestion
    "sourceReference": "https://qudt.org/vocab/unit/DEGREE_API"
```

File: versions/v1/units.json [hunk 3]
```diff
@@ -10494,5 +10494,124 @@
...
+      "api gravity",
+      "grav",
+      "gravity",
+      "gravit",
+      "dAPI",
+      "degree API",
+      "Degree API",
+      "deg API"
+    ],
+    "symbol": "API",
```
Comment (evertoncolling) [inline versions/v1/units.json:10586]:
```suggestion
    "symbol": "°API",
```

File: versions/v1/units.json [hunk 4]
```diff
@@ -10494,5 +10494,124 @@
...
+      "offset": 0.0
+    },
+    "source": "qudt.org",
+    "sourceReference": "https://qudt.org/vocab/unit/HZ"
+  },
+  {
+    "externalId": "gravity-api:api",
+    "name": "API",
+    "quantity": "Gravity_API",
+    "longName": "API Gravity",
```
Comment (evertoncolling) [inline versions/v1/units.json:10571]:
```suggestion
    "longName": "Degree API",
```

File: versions/v1/units.json [hunk 5]
```diff
@@ -10494,5 +10494,124 @@
...
+      "multiplier": 100.0,
+      "offset": 0.0
+    },
+    "source": "qudt.org",
+    "sourceReference": "https://qudt.org/vocab/unit/HZ"
+  },
+  {
+    "externalId": "gravity-api:api",
+    "name": "API",
+    "quantity": "Gravity_API",
```
Comment (evertoncolling) [inline versions/v1/units.json:10570]:
```suggestion
    "quantity": "API Gravity",
```

File: versions/v1/units.json [hunk 6]
```diff
@@ -10494,5 +10494,124 @@
...
+      "API Gravity",
+      "api gravity",
+      "grav",
+      "gravity",
+      "gravit",
+      "dAPI",
+      "degree API",
+      "Degree API",
+      "deg API"
+    ],
```
Comment (evertoncolling) [inline versions/v1/units.json:10585]:
```suggestion
      "deg API",
      "°API"
    ],
```

File: versions/v1/units.json [hunk 7]
```diff
@@ -10494,5 +10494,124 @@
...
+    "symbol": "hHz",
+    "conversion": {
+      "multiplier": 100.0,
+      "offset": 0.0
+    },
+    "source": "qudt.org",
+    "sourceReference": "https://qudt.org/vocab/unit/HZ"
+  },
+  {
+    "externalId": "gravity-api:api",
```
Comment (evertoncolling) [inline versions/v1/units.json:10568]:
```suggestion
    "externalId": "api_gravity:degree_api",
```

File: versions/v1/units.json [hunk 8]
```diff
@@ -10494,5 +10494,124 @@
...
+    "conversion": {
+      "multiplier": 1.0,
+      "offset": 0.0
+    },
+    "source": "qudt.org",
+    "sourceReference": "https://qudt.org/vocab/quantitykind/Gravity_API"
+  },
+  {
+    "externalId": "gravity-api:kilo-api",
+    "name": "kAPI",
```
Comment (evertoncolling) [inline versions/v1/units.json:10596]:
```suggestion
    "name": "k °API",
```

File: versions/v1/units.json [hunk 9]
```diff
@@ -10494,5 +10494,124 @@
...
+    "quantity": "Gravity_API",
```
Comment (evertoncolling) [inline versions/v1/units.json:10597]:
```suggestion
    "quantity": "API Gravity",
```

File: versions/v1/units.json [hunk 10]
```diff
@@ -10494,5 +10494,124 @@
...
+    "symbol": "API",
+    "conversion": {
+      "multiplier": 1.0,
+      "offset": 0.0
+    },
+    "source": "qudt.org",
+    "sourceReference": "https://qudt.org/vocab/quantitykind/Gravity_API"
+  },
+  {
+    "externalId": "gravity-api:kilo-api",
```
Comment (evertoncolling) [inline versions/v1/units.json:10595]:
```suggestion
    "externalId": "api_gravity:k__api",
```

File: versions/v1/units.json [hunk 11]
```diff
@@ -10494,5 +10494,124 @@
...
+    "name": "kAPI",
+    "quantity": "Gravity_API",
+    "longName": "Kilo API Gravity",
```
Comment (evertoncolling) [inline versions/v1/units.json:10598]:
```suggestion
    "longName": "Kilo Degree API",
```

File: versions/v1/units.json [hunk 12]
```diff
@@ -10494,5 +10494,124 @@
...
+    "aliasNames": [
+      "kapi",
+      "kAPI",
+      "kAPI gravity",
+      "kAPI Gravity",
+      "kapi gravity",
+      "kgrav",
+      "kgravity",
+      "kgravit"
+    ],
```
Comment (evertoncolling) [inline versions/v1/units.json:10608]:
```suggestion
      "kgravity",
      "kgravit",
      "k °API"
    ],
```

File: versions/v1/units.json [hunk 13]
```diff
@@ -10494,5 +10494,124 @@
...
+    "symbol": "kAPI",
```
Comment (evertoncolling) [inline versions/v1/units.json:10609]:
```suggestion
    "symbol": "k °API",
```

File: versions/v1/units.json [hunk 14]
```diff
@@ -10494,5 +10494,124 @@
...
+    "conversion": {
+      "multiplier": 1000.0,
+      "offset": 0.0
+    },
+    "source": "qudt.org",
+    "sourceReference": "https://qudt.org/vocab/quantitykind/Gravity_API"
```
Comment (evertoncolling) [inline versions/v1/units.json:10615]:
```suggestion
    "source": "Custom based on qudt.org - https://qudt.org/vocab/unit/DEGREE_API",
    "sourceReference": null
```

File: versions/v1/units.json [hunk 15]
```diff
@@ -10494,5 +10494,124 @@
...
+    "conversion": {
+      "multiplier": 100.0,
+      "offset": 0.0
+    },
+    "source": "SPE Petrowiki",
+    "sourceReference": "https://petrowiki.spe.org/Sucker-rod_pumping_units"
+  },
+  {
+    "externalId": "frequency:hecto-hz",
+    "name": "hecto-HZ",
```
Comment (evertoncolling) [inline versions/v1/units.json:10546]:
```suggestion
    "name": "HectoHZ",
```

File: versions/v1/units.json [hunk 16]
```diff
@@ -10494,5 +10494,124 @@
...
+    "symbol": "spm",
+    "conversion": {
+      "multiplier": 100.0,
+      "offset": 0.0
+    },
+    "source": "SPE Petrowiki",
+    "sourceReference": "https://petrowiki.spe.org/Sucker-rod_pumping_units"
+  },
+  {
+    "externalId": "frequency:hecto-hz",
```
Comment (evertoncolling) [inline versions/v1/units.json:10545]:
```suggestion
    "externalId": "frequency:hectohz",
```

File: versions/v1/units.json [hunk 17]
```diff
@@ -10494,5 +10494,124 @@
...
+      "strokes/m",
+      "Strokes/m"
+    ],
+    "symbol": "spm",
+    "conversion": {
+      "multiplier": 1.0,
+      "offset": 0.0
+    },
+    "source": "SPE Petrowiki",
+    "sourceReference": "https://petrowiki.spe.org/Sucker-rod_pumping_units"
```
Comment (evertoncolling) [inline versions/v1/units.json:10521]: This is not a valid reference. The page has no mention to strokes per minute, nor to conversion factors. Valid references are standards like [qudt.org](qudt.org), [Energistics](https://energistics.org/energistics-unit-of-measure-standard), [OPC-UA/UNECE](https://unece.org/trade/documents/2021/06/uncefact-rec20-0), or publications (see unit with `externalId = "energy:boe"`).

**Review by evertoncolling — CHANGES_REQUESTED** [review]
Thank you for your contribution to the Cognite Unit Catalog. Before we can approve this addition, a few changes are necessary:
- For SPM and HectoSPM: Please provide a trustworthy reference to verify these units.
- For the remaining units: I've left inline comments with specific suggestions. Please address these issues in your next update.

Once you've made these changes, feel free to request another review.

**Comment by evertoncolling** [comment]
> Thanks @evertoncolling for your review. It has cleared many questions that I had. :)
> 
> I have changed a few things further.

Everything seems to be correct now with regards to the `units.json` file. Since you introduced a new quantity (API Gravity), we also need to modify the `unitSystems.json` file, specifying a default unit for the new quantity to at least the Default unit system. Once that is done, the PR tests should pass and we should be able to merge the new units.

**Comment by hanish2712** [comment]
Hello @evertoncolling  I've added the change in unitSystems.json as well.

1st test got passed but second one failed with an unknown error.

**Comment by polomani** [comment]
@hanish2712 should be fixed now

**Review by evertoncolling — APPROVED** [review]
Everything looks good!

**Pull request merged**